src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -426,4 +472,115 @@ public void runTaskTimerAsynchronously(@NotNull Plugin plugin, @NotNull Consumer
 		// TODO Auto-generated method stub
 		throw new UnimplementedOperationException();
 	}
+
+	protected int getActiveRunningCount()
+	{
+		return pool.getActiveCount();
+	}
+
+	private static class TaskList
+	{
+
+		private final Map<Integer, ScheduledTask> tasks;
+
+		private TaskList()
+		{
+			tasks = new ConcurrentHashMap<>();
+		}
+
+		/**
+		 * Add a task but locks the Task list to other writes while adding it.
+		 *
+		 * @param task the task to remove.
+		 * @return true on success.
+		 */
+		private boolean addTask(ScheduledTask task)
+		{
+			if (task == null)
+			{
+				return false;
+			}
+			tasks.put(task.getTaskId(), task);
+			return true;
+		}
+
+		/**
+		 * Remove a task by id but locks the Task list to other writes while removing it.
+		 *
+		 */
+		/*@Deprecated
+		protected ScheduledTask removeTask(ScheduledTask task)
+		{
+			if (!tasks.containsKey(task.getTaskId())) {
+				return null;
+			}
+			taskLock.writeLock().lock();
+			ScheduledTask removed = tasks.remove(task.getTaskId());
+			taskLock.writeLock().unlock();
+			return removed;
+
+		}*/
+
+		/**
+		 * Remove a task by id but locks the Task list to other writes while removing it.
+		 * Not generally there is no reason to remove tasks just cancel them.
+		 *
+		 */
+		/*@Deprecated
+		protected ScheduledTask removeTask(int taskID)
+		{
+			taskLock.writeLock().lock();
+			ScheduledTask removed = tasks.remove(taskID);
+			taskLock.writeLock().unlock();
+			return removed;
+
+		}*/
+
+		protected final List<ScheduledTask> getCurrentTaskList()
+		{
+			List<ScheduledTask> out = new ArrayList<>();
+			if (tasks.size() == 0)
+			{
+				return out;
+			}
+			if (out.addAll(tasks.values()))
+			{
+				return out;
+			}
+			else
+			{
+				throw new RuntimeException("Could not get current task list.");
+			}
```

Comment:
Is this ever reached? The only reason I can imagine `addAll` would ever return `false` is if tasks is empty, but you're already checking for this condition earlier in this function.

Reply:
Should be resolved by 69b612b7f159be6b166906cef5479244eff729b4